codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/config/ClientConfig.java

```diff
@@ -186,6 +195,7 @@ public Builder name(String name) {
          */
         public Builder type(String type) {
             this.type = type;
```

Comment:
I think we should remove `type` from the ClientConfig (but leave it in the Builder).  When we construct a `ClientConfig`, we check the builder for `documentationType` and use that if set, otherwise we set `documentationType` to the `builder.type`.  Eg:

```
public ClientConfig(Builder builder) {
  this.documentationType = builder.documentationType != null ? builder.documentationType : builder.type;
```